internal/pkg/cli/svc_delete.go

```diff
@@ -0,0 +1,362 @@
+
+	// Internal state.
+	environments []*config.Environment
+}
+
+func newDeleteSvcOpts(vars deleteSvcVars) (*deleteSvcOpts, error) {
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("intialize workspace client: %w", err)
+	}
+
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	provider := session.NewProvider()
+	defaultSession, err := provider.Default()
+	if err != nil {
+		return nil, err
+	}
+
+	return &deleteSvcOpts{
+		deleteSvcVars: vars,
+
+		ws:      ws,
+		store:   store,
+		spinner: termprogress.NewSpinner(),
+		sess:    provider,
+		appCFN:  cloudformation.New(defaultSession),
+		getSvcCFN: func(session *awssession.Session) svcDeleter {
+			return cloudformation.New(session)
+		},
+		getECR: func(session *awssession.Session) imageRemover {
+			return ecr.New(session)
+		},
+	}, nil
+}
+
+// Validate returns an error if the user inputs are invalid.
+func (o *deleteSvcOpts) Validate() error {
+	if o.AppName() == "" {
+		return errNoAppInWorkspace
+	}
+	if o.Name != "" {
+		if _, err := o.store.GetService(o.AppName(), o.Name); err != nil {
+			return err
+		}
+	}
+	if o.EnvName != "" {
+		if err := o.validateEnvName(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Ask prompts the user for any required flags.
+func (o *deleteSvcOpts) Ask() error {
+	if err := o.askSvcName(); err != nil {
+		return err
+	}
+
+	if o.SkipConfirmation {
+		return nil
+	}
+
+	deleteConfirmed, err := o.prompt.Confirm(
+		fmt.Sprintf(fmtSvcDeleteConfirmPrompt, o.Name, o.appName),
+		svcDeleteConfirmHelp)
+	if err != nil {
+		return fmt.Errorf("svc delete confirmation prompt: %w", err)
+	}
+	if !deleteConfirmed {
+		return errSvcDeleteCancelled
+	}
+	return nil
+}
+
+// Execute deletes the application's CloudFormation stack, ECR repository, SSM parameter, and local file.
+func (o *deleteSvcOpts) Execute() error {
+	if err := o.appEnvironments(); err != nil {
+		return err
+	}
+
+	if err := o.deleteStacks(); err != nil {
+		return err
+	}
+	if err := o.emptyECRRepos(); err != nil {
+		return err
+	}
+	if err := o.removeSvcFromApp(); err != nil {
+		return err
+	}
+	if err := o.deleteSSMParam(); err != nil {
+		return err
+	}
+	if err := o.deleteWorkspaceFile(); err != nil {
+		return err
+	}
+
+	log.Successf("Deleted service %s from application %s.\n", o.Name, o.appName)
+	return nil
+}
+
+func (o *deleteSvcOpts) validateEnvName() error {
+	if _, err := o.targetEnv(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *deleteSvcOpts) targetEnv() (*config.Environment, error) {
+	env, err := o.store.GetEnvironment(o.AppName(), o.EnvName)
+	if err != nil {
+		return nil, fmt.Errorf("get environment %s from config store: %w", o.EnvName, err)
+	}
+	return env, nil
+}
+
+func (o *deleteSvcOpts) askSvcName() error {
+	if o.Name != "" {
+		return nil
+	}
+
+	names, err := o.serviceNames()
+	if err != nil {
+		return err
+	}
+	if len(names) == 0 {
```

Comment:
Can we make use of the function in "selector.go" that @kohidave added before to reduce repeated code across this pkg?

Reply:
THANK YOU

Reply:
I'll add this in the next follow-up PR. 
I think we need to move the `selector` to its own package because its parameters are private structs (ex: `selectServiceRequest`) so I can't create mocks from it unfortunately :P